### PR TITLE
[CI] Bump jenkinsfiles to use rocm6.1

### PR DESCRIPTION
**Pull request by manupak**

### CODE REVIEW

File: mlir/utils/jenkins/Jenkinsfile [hunk 1]
```diff
@@ -71,7 +71,7 @@ String dockerImage() {
 }
```
Comment (jerryyin) [inline mlir/utils/jenkins/Jenkinsfile:71]: I think line 70 also needs change.

**Review by jerryyin — APPROVED** [review]
Could you address the comment as well as adding `mlir/utils/jenkins/Jenkinsfile.release`?

I'm giving approval to reduce back and forth

File: mlir/utils/jenkins/Jenkinsfile (hunk 1 above)
Comment (manupak) [reply mlir/utils/jenkins/Jenkinsfile:71]: done!

**Comment by manupak** [comment]
Thanks for catching that

**Comment by manupak** [comment]
So 6.1 data layout mismatch is still there -- I may have been comparing with a newer rocm that is being used in SDXL.
Additionally rocm 6.1 turns out not to have a clang-format binary shipped with it -- which is another problem.

Created a new PR to solve the issue differently : https://github.com/ROCm/rocMLIR/pull/1498.

**Comment by manupak** [comment]
@jerryyin I ve now switched to base clang-format in our docker for now to get things going.

I think we need to do a bump our Dockerfile to match https://github.com/google/llvm-premerge-checks/blob/main/containers/buildbot-linux/Dockerfile in the next release. I d suggest we do not
do this *now* just for clang-format.

https://github.com/ROCm/rocMLIR-internal/issues/1476.

**Pull request merged**